src/components/Toolbar/Index.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import icons, {
+    images
+} from '../../assets/index';
+
+/*Only one button is permitted on the left
+it must either be a menu button which has a 'burger' icon
+or a back button which has an arrow icon*/
+export enum LeftButtonTypes {
+    menu = "menu",
+    back = "back"
+}
+/*Multiple buttons can be added to the right side of the toolbar
+if an icon isn't specified the title will be displayed in its place*/
+interface RightButtonType {
+    title: string;
+    icon ? : any;
+    action: () => void;
+}
+/*This specifies the toolbar properties we intend to allow the pages to control
+currently the title, left button type (menu or back) and right buttons can be specified*/
+interface ToolbarProps {
+    title: string;
+    leftButtonType: LeftButtonTypes;
+    rightButtons: Array < RightButtonType > ;
+}
+
+export default class Toolbar extends React.Component < ToolbarProps, {} > {
+    constructor(props: ToolbarProps) {
+    super(props);
+    }
+    /*When a button is selected by a user, find its position 
+    in the actions array and call specified function*/
+    onActionSelected = (position: number) => {
+        this.props.rightButtons[position].action();
+    }
+    render() {
+        //array to hold all action (right buttons) for the page
+        let actionArray = new Array < ToolbarAndroidAction > ();
+        /*Iterate through right buttons added to the page
+        build action objects based on the props and push them to the actionArray*/
+        this.props.rightButtons.map(RightButton => {
+            let action: ToolbarAndroidAction = {
+                title: RightButton.title,
+                show: 'always'
+            }
+            if (RightButton.icon) {
+                action.icon = RightButton.icon;
+            }
+            actionArray.push(action)
+        })
+        return ( <
+            ToolbarAndroid title = {
+                this.props.title
+            }
+            titleColor = 'white'
+            style = {
+                styles.toolbar
+            }
+            /*If the left button specified on the page is a 'menu' type
+            use the menu icon otherwise use the 'back' icon*/
+            navIcon = {
+                (this.props.leftButtonType == "menu") ? icons.menu : icons.back
+            }
+            /*If the left button specified on the page is a 'menu' type
+            use the menu function otherwise use the 'back' function*/
+            onIconClicked = {
+                (this.props.leftButtonType == "menu") ? () => console.log("menu icon selected") :
+                    () => console.log("back icon selected")
+            }
+            onActionSelected = {
+                this.onActionSelected
+            }
+            actions = {
+                actionArray
+            }
+            />
+        )
+    }
+}
+
+const styles = StyleSheet.create({
+    toolbar: {
+        height: 56,
+        backgroundColor: '#4883da',
+    },
+});
```

Comment:
Nice way of saying it brad haha. Most text editors have an option to ensure a newline at end of files I would recommend turning it on. Also even though it sounds cosmestic, it kind of isn't because some parsers do break if a new line isn't at the end of file.